src/NUnitEngine/nunit.engine/RuntimeFramework.cs

```diff
@@ -608,24 +627,25 @@ private static void AppendOlderDotNetFrameworkVersion(List<RuntimeFramework> fra
         // microsoft will do in the future
         private static void AppendDotNetFourFrameworkVersions(List<RuntimeFramework> frameworks, RegistryKey versionKey)
         {
-            foreach (string profile in new string[] { "Full", "Client" })
+            foreach (Profile profile in new Profile[] { Profile.Full, Profile.Client })
             {
-                var profileKey = versionKey.OpenSubKey(profile);
+                var profileKey = versionKey.OpenSubKey(profile.ToString());
                 if (profileKey == null) continue;
                 
                 if (CheckInstallDword(profileKey))
                 {
                     var framework = new RuntimeFramework(RuntimeType.Net, new Version(4, 0));
-                    framework.DisplayName += " - " + profile;
+                    framework.DisplayName += " - " + profile.ToString();
+                    framework.Profile = profile;
                     frameworks.Add(framework);
 
                     var release = (int)profileKey.GetValue("Release", 0);
                     if (release > 0)
                     {
                         framework = new RuntimeFramework(RuntimeType.Net, new Version(4, 5));
```

Comment:
@CharliePoole The loop should be broken out of once frameworks have been added? i.e. if full frameworks are installed, it won't check for client. Have I perhaps missed something here - not sure I can see it?

I agree it's not the clearest method really, although again I went with minimising changes - do you think it would be better written out sequentially? For the three frameworks it's not so much duplication, looking at .NET4.6, looks like 4.0 is really the exception, and others follow a standardised patter.